mongodbatlas/data_lakes.go

```diff
@@ -0,0 +1,236 @@
+package mongodbatlas
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+)
+
+const (
+	dataLakesBasePath = "groups"
+)
+
+// DataLakeService is an interface for interfacing with the Data Lake endpoints of the MongoDB Atlas API.
+// See more: https://docs.mongodb.com/datalake/reference/api/datalakes-api
+type DataLakeService interface {
+	List(context.Context, *DataLakeReqPathParameters, *ListOptions) ([]DataLake, *Response, error)
+	Get(context.Context, *DataLakeReqPathParameters) (*DataLake, *Response, error)
+	Create(context.Context, *DataLakeReqPathParameters, *DataLakeCreateRequest) (*DataLake, *Response, error)
+	Update(context.Context, *DataLakeReqPathParameters, *DataLakeUpdateRequest) (*DataLake, *Response, error)
+	Delete(context.Context, *DataLakeReqPathParameters) (*Response, error)
+}
+
+// DataLakeServiceOp handles communication with the DataLakeService related methods of the
+// MongoDB Atlas API
+type DataLakeServiceOp service
+
+type AwsCloudProviderConfig struct {
```

Comment:
[nit] For public structs we try to keep them documented (some older versions of go vet used to complain about it)
```suggestion
// AwsCloudProviderConfig is the ...
type AwsCloudProviderConfig struct {
```
Same for all other exported structs